emacs/merlin.el

```diff
@@ -1316,15 +1316,18 @@ If QUIET is non nil, then an overlay and the merlin types can be used."
       (set-temporary-overlay-map merlin-type-enclosing-map t
                                  'merlin--type-enclosing-reset))))
 
-(defun merlin-type-enclosing ()
+(defun merlin-type-enclosing (&optional manual)
   "Print the type of the expression under point (or of the region, if it exists).
-If called repeatedly, increase the verbosity of the type shown."
-  (interactive)
-  (if (region-active-p)
-      (merlin--type-region)
-    (when (merlin--type-enclosing-query)
-      (merlin-type-enclosing-go-up)
-      (merlin--type-enclosing-after))))
+If called repeatedly, increase the verbosity of the type shown.
+With prefix argument MANUAL, call `merlin-type-expr' interactively."
+  (interactive "P")
+  (if manual
+      (call-interactively #'merlin-type-expr)
+    (if (region-active-p)
+        (merlin--type-region)
+      (when (merlin--type-enclosing-query)
+        (merlin-type-enclosing-go-up)
+        (merlin--type-enclosing-after)))))
```

Comment:
These changes looks out-of-place in this PR.
Additionally I don't see the benefit of thi new argument instead of calling directly `merlin-type-expr` ?

Reply:
@voodoos I don't think so, this follows the [initial suggestion by @Chris00](https://github.com/ocaml/merlin/issues/1386#issue-993828239):

> Contrarily to [what is asserted on the wiki](https://github.com/ocaml/merlin/wiki/emacs-from-scratch), `C-c t` is no longer bound to `merlin-type-expr` (a [good thing as it is reserved to the user](https://www.gnu.org/software/emacs/manual/html_node/elisp/Key-Binding-Conventions.html)). May we suggest to bind it to `C-c C-t` with a prefix?

To be more precise, @Chris00 meant:

* *Keeping the `C-c C-t` binding as is for `merlin-type-enclosing`* (so, no backward-compatibility issue),
* and *bind `C-u C-c C-t` to `merlin-type-expr`*.

Given `C-u` is the universal prefix, TTBOMK this latter binding should be implemented the way I did: accordingly it's a bit verbose in the diff, but I think this code is idiomatic (otherwise @monnier would have told otherwise I guess :)